provider/runbook_resource_test.go

```diff
@@ -333,98 +391,134 @@ func testAccCheckRunbookResourceDestroy() resource.TestCheckFunc {
       channel_name_format = "-inc-{{ number }}"
-    }
+    })
   }
 }`, rName)
 }
 
 func testAccRunbookResourceConfig_update(rName string) string {
 	return fmt.Sprintf(`
 resource "firehydrant_team" "test_team1" {
-	name = "test-team1-%s"
+  name = "test-team1-%s"
 }
+
 data "firehydrant_runbook_action" "notify_channel" {
-	slug             = "notify_channel"
-	integration_slug = "slack"
-	type             = "incident"
+  slug             = "notify_channel"
+  integration_slug = "slack"
+  type             = "incident"
+}
+
+data "firehydrant_runbook_action" "archive_channel" {
+  slug             = "archive_incident_channel"
+  integration_slug = "slack"
+  type             = "incident"
 }
 
 resource "firehydrant_runbook" "test_runbook" {
-	name        = "test-runbook-%s"
-	type        = "incident"
-	description = "test-description-%s"
-	owner_id    = firehydrant_team.test_team1.id
-
-	steps {
-		name             = "Notify Channel"
-		action_id        = data.firehydrant_runbook_action.notify_channel.id
-		repeats          = true
-		repeats_duration = "PT15M"
-		config = {
-			"channels" = "#incidents"
-		}
-	}
+  name        = "test-runbook-%s"
+  type        = "incident"
+  description = "test-description-%s"
+  owner_id    = firehydrant_team.test_team1.id
+
+  steps {
+    name             = "Notify Channel"
+    action_id        = data.firehydrant_runbook_action.notify_channel.id
+    repeats          = true
+    repeats_duration = "PT15M"
+
+    config = jsonencode({
+      channels = "#incidents"
+    })
+  }
+
+  steps {
+    name      = "Archive Channel"
+    action_id = data.firehydrant_runbook_action.archive_channel.id
+  }
 }`, rName, rName, rName)
 }
 
-func testAccRunbookResourceConfig_required_repeat_duration_not_set(rName string) string {
+func testAccRunbookResourceConfig_stepsRequiredRepeatsDurationNotSet(rName string) string {
 	return fmt.Sprintf(`
 data "firehydrant_runbook_action" "create_incident_channel" {
-	slug             = "create_incident_channel"
-	integration_slug = "slack"
-	type             = "incident"
+  slug             = "create_incident_channel"
+  integration_slug = "slack"
+  type             = "incident"
 }
 
 resource "firehydrant_runbook" "test_runbook" {
-	name = "test-runbook-%s"
-	type = "incident"
-
-	steps {
-		name      = "Create Incident Channel"
-		repeats   = true
-		action_id = data.firehydrant_runbook_action.create_incident_channel.id
-		config = {
-			channel_name_format = "-inc-{{ number }}"
-		}
-	}
+  name = "test-runbook-%s"
+  type = "incident"
+
+  steps {
+    name      = "Create Incident Channel"
+    repeats   = true
+    action_id = data.firehydrant_runbook_action.create_incident_channel.id
+
+    config = jsonencode({
+      channel_name_format = "-inc-{{ number }}"
+    })
+  }
 }`, rName)
 }
 
-func testAccRunbookResourceConfig_required_repeats_not_set(rName string) string {
+func testAccRunbookResourceConfig_stepsRequiredRepeatsNotSet(rName string) string {
```

Comment:
Ah so thats the casing convention, I was really thrown off by why have underscore and camelcase in places

Reply:
i think it is but honestly, i've seen things be a lil different across multiple go repos, which makes things very confusing. this is what they seem to do most of the time in the aws and google providers, which is why i try to do it this way. but we could also kinda do whatever we want i think, as long as it's consistent